### PR TITLE
Refactor native image config to avoid warnings

### PR DESCRIPTION
**Pull request by graemerocher**
This eliminates the warnings produced by the native image Maven plugin which look like:

```
[WARNING] jar:file:///Users/graemerocher/.m2/repository/io/micronaut/micronaut-runtime/2.2.0.BUILD-SNAPSHOT/micronaut-runtime-2.2.0.BUILD-SNAPSHOT.jar!/META-INF/native-image/io.micronaut/runtime-graal/native-image.properties does not match recommended META-INF/native-image/${groupId}/${artifactId}/native-image.properties layout
```

### CODE REVIEW

File: graal/src/main/java/io/micronaut/graal/reflect/GraalTypeElementVisitor.java [hunk 1]
```diff
@@ -261,16 +259,6 @@ private void generateNativeImageProperties(VisitorContext visitorContext) {
...
-            String propsFile = path + NATIVE_IMAGE_PROPERTIES;
-
-            visitorContext.visitMetaInfFile(propsFile).ifPresent(gf -> {
-                visitorContext.info("Writing " + NATIVE_IMAGE_PROPERTIES + " file to destination: " + gf.getName());
-                try (PrintWriter w = new PrintWriter(gf.openWriter())) {
-                    w.println("Args = -H:ReflectionConfigurationResources=${.}/reflection-config.json");
-                } catch (IOException e) {
-                    visitorContext.fail("Error writing " + NATIVE_IMAGE_PROPERTIES + ": " + e.getMessage(), null);
-                }
-            });
```
Comment (ilopmar) [inline graal/src/main/java/io/micronaut/graal/reflect/GraalTypeElementVisitor.java:273]: Why do you delete this?

Comment (graemerocher) [reply graal/src/main/java/io/micronaut/graal/reflect/GraalTypeElementVisitor.java:273]: if the file is named `reflect-config.json` it is automatically picked up. We were only generating this file to load our inconsistently named `reflection-config.json` file

Comment (ilopmar) [reply graal/src/main/java/io/micronaut/graal/reflect/GraalTypeElementVisitor.java:273]: You are right. Then I think you can rename `REFLECTION_CONFIG_JSON` to `REFLECT_CONFIG_JSON` here https://github.com/micronaut-projects/micronaut-core/pull/4541/files#diff-a6f6eed2d556fc4571d32b10ec2a16d17cb98075c9e1b3ae0395176281cbb061R73

For the rest of the changes, everything looks good.

Comment (graemerocher) [reply graal/src/main/java/io/micronaut/graal/reflect/GraalTypeElementVisitor.java:273]: That is already done

**Review by ilopmar — APPROVED** [review]

**Pull request merged**